src/Tasks/Hash.cs

```diff
@@ -49,7 +54,11 @@ public override bool Execute()
                     {
                         foreach (var item in ItemsToHash)
                         {
-                            stringBuilder.Append(item.ItemSpec);
+                            foreach (char c in item.ItemSpec)
+                            {
+                                stringBuilder.Append(IgnoreCase ? char.ToLower(c) : c);
```

Comment:
Should this use `ToLowerInvariant()` instead?

Reply:
Should we be worried about the perf implications of this?

Reply:
Also, use `ToUpperInvariant()`. I believe some characters don't convert correctly to lower so ToUpper is preferred in cases like this.